python/django_bolt/auth/tokens.py

```
@@ -0,0 +1,300 @@
+        # family so the legitimate holder's chain dies too (reuse detection).
+        if rotate and fam is not None:
+            await _revoke_family(store, fam, exp=refresh_claims.get("exp"))
+        raise TokenRotationError("Refresh token revoked or already used")
+
+    oat = refresh_claims.get("oat")
+    if (
+        max_session_lifetime is not None
+        and oat is not None
+        and int(time.time()) - int(oat) > max_session_lifetime
+    ):
+        raise TokenRotationError("Session exceeded maximum lifetime")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)